src/main/java/sorald/DefaultRepair.java

```diff
@@ -14,101 +14,107 @@
-      Factory factory = launcher.getFactory();
-      ProcessingManager processingManager = new QueueProcessingManager(factory);
-      processingManager.addProcessor(processor);
-      JavaOutputProcessor javaOutputProcessor = launcher.createOutputWriter();
-
-      if (this.config.getFileOutputStrategy() == FileOutputStrategy.ALL
-          || outputDirPath.contains(intermediateSpoonedPath)) {
-        processingManager.addProcessor(javaOutputProcessor);
-      }
-      processingManager.process(factory.Class().getAll());
-
-      if (this.config.getFileOutputStrategy() == FileOutputStrategy.CHANGED_ONLY
-          && !outputDirPath.contains(intermediateSpoonedPath)) {
-        for (Map.Entry<String, CtType> patchedFile :
-            UniqueTypesCollector.getInstance().getTopLevelTypes4Output().entrySet()) {
-          javaOutputProcessor.process(patchedFile.getValue());
-          if (this.config.getGitRepoPath() != null) {
-            createPatches(patchedFile.getKey(), javaOutputProcessor);
-          }
-        }
-      }
+    private List<SoraldAbstractProcessor> addedProcessors = new ArrayList();
+
+    public DefaultRepair(SoraldConfig config) {
+        super(config);
     }
 
-    this.printEndProcess();
-    deleteDirectory(new File(intermediateSpoonedPath));
-    UniqueTypesCollector.getInstance().reset();
-  }
+    public void repair() {
+        final String spoonedPath = this.config.getWorkspace() + File.separator + Constants.SPOONED;
+        final String intermediateSpoonedPath =
+                spoonedPath + File.separator + Constants.INTERMEDIATE;
+
+        String inputDirPath;
+        String outputDirPath;
+
+        List<Integer> ruleKeys = this.config.getRuleKeys();
+        for (int i = 0; i < ruleKeys.size(); i++) {
+            int ruleKey = ruleKeys.get(i);
+
+            if (ruleKeys.size()
+                    == 1) { // one processor, straightforward repair: we use the given input file
+                // dir, run one
+                // processor, directly output files (independently of the FileOutputStrategy) in
+                // the final output dir
+                inputDirPath = this.config.getOriginalFilesPath();
+                outputDirPath = spoonedPath;
+            } else { // more than one processor, thus we need an intermediate dir, which will always
+                // contain all files (the changed and non-changed ones), because other processors
+                // will run on them
+                if (i == 0) { // the first processor will run, thus we use the given input file
+                    // dir and output
+                    // *all* files in the intermediate dir
+                    inputDirPath = this.config.getOriginalFilesPath();
+                    outputDirPath = intermediateSpoonedPath;
+                } else if ((i + 1)
+                        == ruleKeys.size()) { // the last processor will run, thus we use as input
```

Comment:
The formatter does this to limit line lengths when there's an inline comment that makes the line too long. It looks pretty funky. We should probably try to avoid inline comments like this one :)